src/demo/demo.js

```diff
@@ -58,39 +50,42 @@ class SDK extends Component {
         process.env.NODE_ENV
       )
     }
+
     console.log('Calling `Onfido.init` with the following options:', options)
 
-    const onfidoSdk = Onfido.init({ ...options, containerEl: this.el })
-    this.setState({ onfidoSdk })
+    if (mountEl.current) {
+      const sdk = Onfido.init({ ...options, containerEl: mountEl.current })
+      setOnfidoSdk(sdk)
 
-    window.onfidoSdkHandle = onfidoSdk
-  }
+      window.onfidoSdkHandle = onfidoSdk
+    }
 
-  shouldComponentUpdate() {
-    return false
-  }
-  render = () => <div ref={(el) => (this.el = el)} />
-}
+    return () => onfidoSdk && onfidoSdk.tearDown()
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
What is this second argument here for? I can't see it in the `useEffect` docs

Reply:
It's in the note of this section: https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects. To sum-up:
- No dependencies array: the effect will run for every re-render
- Empty dependencies array: the effect will run once when the component mounted and the clean-up function (the return inside the effect) will be run once when the component unmounts.
- Non-empty dependencies array: the effect will run every time the dependencies change, just like `componentDidUpdate` with relevant prop name handles. They're still checked by reference, not value, i.e. the new object with the same values (recursively) but initialised independently will be treated as an update.

Reply:
Oh, I see! Not sure how I missed, thanks for clarifying that.